### PR TITLE
Todo branch

### PR DESCRIPTION
**Pull request by Oxford-G**
This project is a javascript-based Todo web application.
 Features include the following
- Filterale todos based on three parameters (All Todos, Pending Todos, Completed Todos).
- User can delete the todos.
- Highlight todos by toggling the checkbox.

### CODE REVIEW

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Changes required❌

# 👋 @Oxford-G 

#### You have done a good job on this one so far, there is default project, users can read, update, delete todos and projects, and Webpack is well configured🤗 .

Here are some issues to address to get this approved:

* [ ] It uses ES6 syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/))
- Let us use arrow functions in all our function definitions

* [ ] Let us state explicitly on which port/url to start the application after `npm start`.

* [ ] Do not modify linter configuration, let us remove the comments and try to

**[Optional]**

* [ ] It would be better not to allow the deletion of  default projects, the others however, can be deleted. The default one should be such that we can remove the it `todos` but the project name should always be available.

* [ ] It is always better to capitalize your commit messages.

* [ ] We can improve on the UI with some styling, It could be a good candidate for our future portfolio.

#### That is it kindly make the changes and submit a re-review😎

#### Happy coding😇

Addo Forison

File: src/project.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// eslint-disable-next-line import/no-cycle
```
Comment (Forison) [inline src/project.js:1]: * [ ]  Do not modify linter configurations
Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/) and [Zendesk](https://microverse.zendesk.com/hc/en-us/articles/1500002085202-February-2nd-2021-linter-errors-as-a-reason-of-project-invalidation-)))

File: src/project.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+class Project {
+  constructor(name) {
+    this.name = name;
+    this.id = name.toLowerCase();
+    this.todos = [];
+  }
+}
+
+const projectModule = (() => {
+  function createProject(target) {
```
Comment (Forison) [inline src/project.js]: Let us use arrow functions in function definition.

File: src/start.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+import project, { Project, removeProject } from './project';
+// eslint-disable-next-line import/no-cycle
+import {
+  Todo, createTodo, removeTodo, updateStatus,
+} from './task';
+
+const container = document.getElementById('content');
+
+const predefinedProjects = [new Project('Default'), new Project('Yesterday'), new Project('Today'), new Project('Tomorrow')];
+function getProjects() {
```
Comment (Forison) [inline src/start.js]: Let us use arrow functions in function definition.

File: src/start.js [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+      for (let i = 0; i < todoList.length; i += 1) {
+        Object.setPrototypeOf(todoList[i], Todo.prototype);
+      }
+    }
+    return projects;
+  }
+  return predefinedProjects;
+}
+
+function clearContent(element) {
```
Comment (Forison) [inline src/start.js]: Let us use arrow functions in function definition.
> There are other areas that require the same changes, let us address all of those.

File: src/start.js [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+  element.textContent = '';
+}
+
+function displayProjects() {
```
Comment (Forison) [inline src/start.js:31]: Let us use arrow functions in function definition.

File: src/start.js [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+      // eslint-disable-next-line no-use-before-define
+      displayTodos(selectedProjectId);
+    });
+  });
+  // eslint-disable-next-line no-use-before-define
+  displayTodos(selectedProjectId);
+}
+
+function displayTodoForm() {
```
Comment (Forison) [inline src/start.js:53]: Let us use arrow functions in function definition.

File: src/start.js [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+    createTodo(title, date, description, priority, project);
+  });
+
+  const cancelTodoBtn = buttons.appendChild(document.createElement('button'));
+  cancelTodoBtn.setAttribute('class', 'btn btn-danger cancel-todo');
+  cancelTodoBtn.innerHTML = 'Cancel';
+  cancelTodoBtn.addEventListener('click', displayProjects);
+}
+
+function displayTodos() {
```
Comment (Forison) [inline src/start.js:141]: Let us use arrow functions in function definition.

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED 🔴 🛑 
-----------
Hello @Oxford-G, great job you are doing on this project 👏 
However, there is a minor change to make to complete this milestone 🛠️ see it below 👀 👇 
------------------

- [ ] As the previous reviewer asks you to change, use arrow functions instead of the function keyword.  Change all the functions to arrow functions you still have some that did not change. In `inde.js` and `satrt.js` 
------ 
Please make this required change and send it again for another review 💪 👍 
HAPPY CODING!!! 😄 

kvUU7

**Review by rukundoeric — APPROVED** [review]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 
  - All issues raised from the previous review are addressed. 💯 


- ### To Improve
  - n/a

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/),
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**